query/src/sql/exec/data_schema_builder.rs

```diff
@@ -54,9 +57,10 @@ impl<'a> DataSchemaBuilder<'a> {
     pub fn build_project(
         &self,
         plan: &ProjectPlan,
-        input_schema: DataSchemaRef,
+        _input_schema: DataSchemaRef,
     ) -> Result<DataSchemaRef> {
-        let mut fields = input_schema.fields().clone();
+        // let mut fields = input_schema.fields().clone();
+        let mut fields = vec![];
```

Comment:
Maybe `Vec::with_capacity(plan.items.len())` is better.

Reply:
Fixed.